File: csharp/1701.average-waiting-time.cs
/*
 * @lc app=leetcode id=1701 lang=csharp
 *
 * [1701] Average Waiting Time
 */

// @lc code=start
public class Solution {
    public double AverageWaitingTime(int[][] customers) {
        int t = 0;
        double total = 0;
        foreach (var c in customers) {
            int a = c[0], d = c[1];
            if (t <= a) {
                total += d;
                t = a + d;
            } else {
                total += t - a + d;
                t += d;
            }
        }
        return total / customers.Length;
    }
}
// @lc code=end

File: java/1701.average-waiting-time.java
/*
 * @lc app=leetcode id=1701 lang=java
 *
 * [1701] Average Waiting Time
 */

// @lc code=start
class Solution {
    public double averageWaitingTime(int[][] customers) {
        int n = customers.length, t = customers[0][0];
        double total = 0;
        for (int i = 0; i < n; ++i) {
            var c = customers[i];
            if (t <= c[0]) {
                total += c[1];
                t = c[0] + c[1];
            } else {
                total += c[1] + t - c[0];
                t += c[1];
            }
        }
        return total / n;
    }
}
// @lc code=end

